apps/ui/src/ui/form/OptionField.tsx

```diff
@@ -27,6 +27,9 @@ export default function OptionField<X extends FieldValues, P extends FieldPath<X
         value = field.value
         onChange = field.onChange
     }
+    useEffect(() => {
+        if (!value && options[0]) onChange?.(options[0].key as any)
+    }, [options])
```

Comment:
we should probably avoid this kind of pattern, easy to end up in infinite render loops (triggering re-render of a parent from a child component without any user input)

Reply:
better to set defaults at a higher level when initializing state. or if the default changes based on some other value, update this field then too